### PR TITLE
Use redux in react components

### PR DESCRIPTION
**Pull request by Marcraphael12**
The goal of this PR is to make sure that we are following every project's requirement. Have a good review!
### Project requirements
- First of all - make sure that you keep state immutable in reducers!
- Import Redux `Provider` and your store in the main component of your React app.
- Connect your app to the Redux store.
- Make sure that list of books displays books from Redux store.
    - Use `useSelector` to consume the state.
- Make sure that categories page displays categories from Redux store.
    - Use `useSelector` to consume the state.
- In your React component responsible for adding new books, add funcionality for "Add new" button click:
     - Import Redux `useDispatch`.
     - Set the data inputs in the local React state (set *title* and *author*  - NOTE: *categories* and *comments* are NOT part of this step). 
     - Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for removing books, add a funcionality for "Remove" button click:
    - Import Redux `useDispatch`.
    - Dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for categories, add a funcionality for "Check status" button click:
    - Import Redux `useDispatch`.
    - Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

File: src/components/books/book.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+// import components
+import BookProgress from './progress';
+
+// import actions
+import { removeBook } from '../../redux/books/books';
+
+// we use an id as a parameter to identify the book when time comes to remove it
+const Options = ( { id } ) => {
+  const Remove = () => useDispatch(removeBook(id));
```
Comment (keroloussamy) [inline src/components/books/book.js:12]: - [ ] Kindly this way makes errors and not delete the book. You can make it like 

```
const dispatch = useDispatch();

  const Remove = (e) => {
    dispatch(removeBook(id));
  };
```
Here is the error that I got 
image

**Review by keroloussamy — CHANGES_REQUESTED** [review]
Hi @Marcraphael12, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/books/book.js (hunk 1 above)
Comment (Marcraphael12) [reply src/components/books/book.js]: Thanks for notifying me, I wanted to simplify it but as it doesn't work I'll just rewrite it

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**